src/foam/nanos/auth/PermissionedPropertyDAO.js

```diff
@@ -127,26 +127,28 @@ foam.CLASS({
   FObject obj = oldObj.fclone();
   AuthService auth = (AuthService) x.get("auth");
 
-  if ( propertyMap_.containsKey(of) ) {
-    List properties = propertyMap_.get(of);
-    Iterator e = properties.iterator();
-    while ( e.hasNext() ) {
-      PropertyInfo axiom = (PropertyInfo) e.next();
-      maybeRemove(axiom, of, auth, x, obj, oldObj);
-    }
-  } else {
+  if ( ! propertyMap_.containsKey(of) ) {
     List<PropertyInfo> properties = new ArrayList<>();
     List list = oldObj.getClassInfo().getAxiomsByClass(PropertyInfo.class);
     Iterator e = list.iterator();
     while ( e.hasNext() ) {
       PropertyInfo axiom = (PropertyInfo) e.next();
       if ( axiom.getPermissionRequired() ) {
-        properties.add(axiom);
-        maybeRemove(axiom, of,auth, x, obj, oldObj);
+        boolean authCheck = auth.check(x, of + ".rw." + axiom.getName().toLowerCase()) || auth.check(x, of + ".ro." + axiom.getName().toLowerCase());
+        if ( ! authCheck ) {
+          properties.add(axiom);
+        }
       }
     }
     propertyMap_.put(of, properties);
   }
+    
+  List properties = propertyMap_.get(of);
+  Iterator e = properties.iterator();
+  while ( e.hasNext() ) {
+    PropertyInfo axiom = (PropertyInfo) e.next();
+    axiom.clear(obj);
```

Comment:
I don't think this makes sense. We don't just want to clear every permissioned property every time, we only want to clear it if they don't have the appropriate permission.

Reply:
Right idea, but in wrong place. Move to HidePropertiesSink so that it is only cached/reused for a single select(), and doesn't leak between users.